test/unit/lists-component.spec.ts

```diff
@@ -552,37 +553,118 @@ describe('when creating a new list', () => {
+    beforeEach(() => {
+      // Insert List Mock Query
+      dbClientQueryMock.mockResolvedValueOnce({ rows: [{ id: listId }] })
+
+      // Access Mock Query
+      dbClientQueryMock.mockRejectedValueOnce(new Error("Unexpected error when inserting the list's data"))
+    })
+
+    it('should throw a generic error', async () => {
+      await expect(listsComponent.addList({ name, userAddress, private: false })).rejects.toEqual(new Error("The list couldn't be created"))
+    })
+  })
+
   describe('and there are no lists with the same name', () => {
     let dbList: DBList
     let result: DBList
 
-    beforeEach(async () => {
-      name = 'Test List'
+    beforeEach(() => {
       dbList = {
         id: listId,
         name,
         user_address: userAddress,
         description: null,
         created_at: new Date()
       }
-      dbQueryMock.mockResolvedValueOnce({
+
+      // Create List Query
+      dbClientQueryMock.mockResolvedValueOnce({
         rowCount: 1,
         rows: [dbList]
       })
-      result = await listsComponent.addList({ name, userAddress, private: false })
     })
 
-    it('should create the pick', () => {
-      expect(dbQueryMock).toHaveBeenCalledWith(
-        expect.objectContaining({
-          strings: expect.arrayContaining([expect.stringContaining('INSERT INTO favorites.lists (name, description, user_address)')]),
-          values: [name, null, userAddress]
-        })
-      )
+    describe('and the list should be private', () => {
+      beforeEach(async () => {
+        // Access Mock Query
+        dbClientQueryMock.mockResolvedValueOnce({ rowCount: 1 })
+
+        result = await listsComponent.addList({ name, userAddress, private: true })
+      })
+
+      it('should create the list', () => {
+        expect(dbClientQueryMock).toHaveBeenCalledWith(
+          expect.objectContaining({
+            strings: expect.arrayContaining([expect.stringContaining('INSERT INTO favorites.lists (name, description, user_address)')]),
+            values: [name, null, userAddress]
+          })
+        )
+      })
+
+      it('should not delete any access becasue it is a new list without previous rows in the db', () => {
+        expect(dbClientQueryMock).not.toHaveBeenCalledWith(
+          expect.objectContaining({
+            strings: expect.arrayContaining([
+              expect.stringContaining('DELETE FROM favorites.acl USING favorites.lists'),
+              expect.stringContaining('WHERE favorites.acl.list_id = favorites.lists.id'),
+              expect.stringContaining('AND favorites.acl.list_id ='),
+              expect.stringContaining('AND favorites.lists.user_address ='),
+              expect.stringContaining('AND favorites.acl.permission ='),
+              expect.stringContaining('AND favorites.acl.grantee =')
+            ]),
+            values: [listId, userAddress, Permission.VIEW, '*']
+          })
+        )
+      })
```

Comment:
What if we assert here that the the insert of the row in the ACL was not performed? I'm saying this because the `addList` function never performs a `DELETE`, but it could accidentally perform an INSERT into the ACL table.
```suggestion
      it('should not insert a new access to make the list public', () => {
        expect(dbClientQueryMock).not.toHaveBeenCalledWith(
          expect.objectContaining({
            strings: expect.arrayContaining([
              expect.stringContaining('INSERT INTO favorites.acl (list_id, permission, grantee) VALUES'),
            ])
          })
        )
      })
```